src/util/matchToKartoffel.js

```diff
@@ -221,6 +221,72 @@ const match_ads = (obj) => {
     })
 };
 
+const match_adNN = (obj) => {
+    const objKeys = Object.keys(obj);
+    objKeys.map((rawKey) => {
+        switch (rawKey) {
+            //firstName
+            case fn.adNN.firstName:
+                obj.firstName = obj[rawKey];
+                (rawKey === "firstName") ? null : delete obj[rawKey];
+                break;
+            //lastName
+            case fn.adNN.lastName:
+                obj.lastName = obj[rawKey];
+                (rawKey === "lastName") ? null : delete obj[rawKey];
+                break;
+            //mail
+            case fn.adNN.mail:
+                obj.mail = obj[rawKey];
+                (rawKey === "mail") ? null : delete obj[rawKey];
+                break;
+            //hierarchy and job
+            case fn.adNN.hierarchy:
+                let hr = obj[rawKey].includes("\\") ? obj[rawKey].substring(0, obj[rawKey].lastIndexOf('\\')).trim().split('\\') : obj[rawKey].substring(0, obj[rawKey].lastIndexOf('/')).trim().split('/');
+                if (hr[0] == "") {
+                    delete obj[rawKey];
+                    break;
+                }
+                hr[0] === fn.rootHierarchy ? null : hr.unshift(fn.rootHierarchy);
+                hr.splice((hr.length - 1), 1);
```

Comment:
remove this line baecause of using lastIndex command

Reply:
removed